diff -urN a/components/cbor/BUILD.gn b/components/cbor/BUILD.gn
--- a/components/cbor/BUILD.gn	2026-10-05 18:48:49.435763886 +0200
+++ b/components/cbor/BUILD.gn	2026-10-05 19:05:55.166721464 +0200
@@ -12,7 +12,7 @@
   # TODO(crbug.com/535682335): Remove `USE_CBOR_RUST` buildflag entirely,
   # unconditionally enable Rust CBOR parser, and drop `is_cronet_build` check
   # once Cronet supports Crubit dependencies.
-  flags = [ "USE_CBOR_RUST=!$is_cronet_build" ]
+  flags = [ "USE_CBOR_RUST=$enable_cpp_api_from_rust" ]
 }
 
 component("cbor") {
@@ -35,7 +35,7 @@
     "//base",
   ]
 
-  if (!is_cronet_build) {
+  if (enable_cpp_api_from_rust) {
     public_deps = [
       "//build/rust/crubit",
       "//components/cbor/rust:cbor_rust_bindings",
@@ -62,7 +62,7 @@
     "//testing/gtest",
     "//third_party/fuzztest",
   ]
-  if (!is_cronet_build) {
+  if (enable_cpp_api_from_rust) {
     deps += [ "//components/cbor/rust:cbor_rust_unittests" ]
   }
 
diff -urN a/components/private_verification_tokens/common/athm_ffi/cxx_ffi.rs b/components/private_verification_tokens/common/athm_ffi/cxx_ffi.rs
--- a/components/private_verification_tokens/common/athm_ffi/cxx_ffi.rs	1970-01-01 01:00:00.000000000 +0100
+++ b/components/private_verification_tokens/common/athm_ffi/cxx_ffi.rs	2026-10-05 18:54:33.855215755 +0200
@@ -0,0 +1,161 @@
+use alloc::{boxed::Box, vec::Vec};
+
+use crate::{AthmClientRequest, AthmKeyMaterial, AthmParameters, AthmStatus, TokenRequest};
+
+#[cxx::bridge(namespace = "private_verification_tokens")]
+mod ffi {
+    enum AthmStatus {
+        InvalidInput,
+        OperationFailed,
+    }
+
+    extern "Rust" {
+        type CxxAthmKeyMaterial;
+        type CxxAthmParameters;
+        type CxxAthmClientRequest;
+        type CxxTokenRequest;
+
+        fn athm_key_material_new(
+            num_buckets: u8,
+            deployment_id: &[u8],
+        ) -> Result<Box<CxxAthmKeyMaterial>, AthmStatus>;
+
+        fn athm_key_material_public_key(
+            key_material: &CxxAthmKeyMaterial,
+        ) -> Vec<u8>;
+
+        fn athm_key_material_public_key_proof(
+            key_material: &CxxAthmKeyMaterial,
+        ) -> Vec<u8>;
+
+        fn athm_key_material_issue(
+            key_material: &CxxAthmKeyMaterial,
+            request: &CxxTokenRequest,
+            hidden_metadata: u8,
+        ) -> Result<Vec<u8>, AthmStatus>;
+
+        fn athm_key_material_verify(
+            key_material: &CxxAthmKeyMaterial,
+            token: &[u8],
+        ) -> Result<u8, AthmStatus>;
+
+        fn athm_parameters_new(
+            num_buckets: u8,
+            deployment_id: &[u8],
+        ) -> Result<Box<CxxAthmParameters>, AthmStatus>;
+
+        fn athm_parameters_token_response_size(
+            params: &CxxAthmParameters,
+        ) -> usize;
+
+        fn athm_client_request_new(
+            public_key: &[u8],
+            public_key_proof: &[u8],
+            params: &CxxAthmParameters,
+        ) -> Result<Box<CxxAthmClientRequest>, AthmStatus>;
+
+        fn athm_client_request_request(
+            request: &CxxAthmClientRequest,
+        ) -> Vec<u8>;
+
+        fn athm_client_request_finalize(
+            request: &CxxAthmClientRequest,
+            public_key: &[u8],
+            response: &[u8],
+            params: &CxxAthmParameters,
+        ) -> Result<Vec<u8>, AthmStatus>;
+
+        fn token_request_decode(
+            bytes: &[u8],
+        ) -> Result<Box<CxxTokenRequest>, AthmStatus>;
+    }
+}
+
+struct CxxAthmKeyMaterial(AthmKeyMaterial);
+struct CxxAthmParameters(AthmParameters);
+struct CxxAthmClientRequest(AthmClientRequest);
+struct CxxTokenRequest(TokenRequest);
+
+fn athm_key_material_new(
+    num_buckets: u8,
+    deployment_id: &[u8],
+) -> Result<Box<CxxAthmKeyMaterial>, AthmStatus> {
+    AthmKeyMaterial::try_new(num_buckets, deployment_id)
+        .map(|value| Box::new(CxxAthmKeyMaterial(value)))
+}
+
+fn athm_key_material_public_key(
+    key_material: &CxxAthmKeyMaterial,
+) -> Vec<u8> {
+    key_material.0.public_key()
+}
+
+fn athm_key_material_public_key_proof(
+    key_material: &CxxAthmKeyMaterial,
+) -> Vec<u8> {
+    key_material.0.public_key_proof()
+}
+
+fn athm_key_material_issue(
+    key_material: &CxxAthmKeyMaterial,
+    request: &CxxTokenRequest,
+    hidden_metadata: u8,
+) -> Result<Vec<u8>, AthmStatus> {
+    key_material.0.issue(&request.0, hidden_metadata)
+}
+
+fn athm_key_material_verify(
+    key_material: &CxxAthmKeyMaterial,
+    token: &[u8],
+) -> Result<u8, AthmStatus> {
+    key_material.0.verify(token)
+}
+
+fn athm_parameters_new(
+    num_buckets: u8,
+    deployment_id: &[u8],
+) -> Result<Box<CxxAthmParameters>, AthmStatus> {
+    AthmParameters::try_new(num_buckets, deployment_id)
+        .map(|value| Box::new(CxxAthmParameters(value)))
+}
+
+fn athm_parameters_token_response_size(
+    params: &CxxAthmParameters,
+) -> usize {
+    params.0.token_response_size()
+}
+
+fn athm_client_request_new(
+    public_key: &[u8],
+    public_key_proof: &[u8],
+    params: &CxxAthmParameters,
+) -> Result<Box<CxxAthmClientRequest>, AthmStatus> {
+    AthmClientRequest::try_new(
+        public_key,
+        public_key_proof,
+        &params.0,
+    )
+    .map(|value| Box::new(CxxAthmClientRequest(value)))
+}
+
+fn athm_client_request_request(
+    request: &CxxAthmClientRequest,
+) -> Vec<u8> {
+    request.0.request()
+}
+
+fn athm_client_request_finalize(
+    request: &CxxAthmClientRequest,
+    public_key: &[u8],
+    response: &[u8],
+    params: &CxxAthmParameters,
+) -> Result<Vec<u8>, AthmStatus> {
+    request.0.finalize(public_key, response, &params.0)
+}
+
+fn token_request_decode(
+    bytes: &[u8],
+) -> Result<Box<CxxTokenRequest>, AthmStatus> {
+    TokenRequest::decode(bytes)
+        .map(|value| Box::new(CxxTokenRequest(value)))
+}
diff -urN a/components/web_package/signed_web_bundles/rust/cxx_ffi.rs b/components/web_package/signed_web_bundles/rust/cxx_ffi.rs
--- a/components/web_package/signed_web_bundles/rust/cxx_ffi.rs	1970-01-01 01:00:00.000000000 +0100
+++ b/components/web_package/signed_web_bundles/rust/cxx_ffi.rs	2026-10-05 18:58:29.302056288 +0200
@@ -0,0 +1,221 @@
+// Copyright 2026 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use alloc::{boxed::Box, string::String, vec::Vec};
+
+use crate::{
+    create_empty_integrity_block_cbor, create_signature_payload,
+    parse_integrity_block, verify_ecdsa_p256_signature,
+    verify_ed25519_signature, Sha512Hasher,
+};
+
+#[cxx::bridge(namespace = "web_package::signed_web_bundles::rust")]
+mod ffi {
+    enum SignatureType {
+        Ed25519,
+        EcdsaP256SHA256,
+        Unknown,
+    }
+
+    extern "Rust" {
+        type IntegrityBlockResult;
+
+        fn err(self: &IntegrityBlockResult) -> bool;
+        fn unwrap(self: &mut IntegrityBlockResult) -> Box<CxxIntegrityBlock>;
+        fn error_message(self: &IntegrityBlockResult) -> String;
+        fn error_is_version_error(self: &IntegrityBlockResult) -> bool;
+
+        type CxxIntegrityBlock;
+
+        fn size(self: &CxxIntegrityBlock) -> usize;
+        fn web_bundle_id(self: &CxxIntegrityBlock) -> String;
+        fn attributes_cbor(self: &CxxIntegrityBlock) -> Vec<u8>;
+        fn signature_stack_size(self: &CxxIntegrityBlock) -> usize;
+        fn signature_stack_entry(
+            self: &CxxIntegrityBlock,
+            index: usize,
+        ) -> Box<CxxSignatureStackEntry>;
+
+        type CxxSignatureStackEntry;
+
+        fn attributes_cbor(self: &CxxSignatureStackEntry) -> Vec<u8>;
+        fn signature_type(self: &CxxSignatureStackEntry) -> SignatureType;
+        fn public_key(self: &CxxSignatureStackEntry) -> Vec<u8>;
+        fn signature(self: &CxxSignatureStackEntry) -> Vec<u8>;
+
+        fn parse_integrity_block_ffi(input: &[u8]) -> Box<IntegrityBlockResult>;
+
+        fn create_empty_integrity_block_cbor(
+            attributes_cbor: &[u8],
+        ) -> Vec<u8>;
+
+        fn create_signature_payload(
+            unsigned_web_bundle_hash: &[u8],
+            integrity_block_cbor: &[u8],
+            attributes_cbor: &[u8],
+        ) -> Vec<u8>;
+
+        fn verify_ed25519_signature(
+            public_key: &[u8],
+            signature: &[u8],
+            message: &[u8],
+        ) -> bool;
+
+        fn verify_ecdsa_p256_signature(
+            public_key: &[u8],
+            signature: &[u8],
+            message: &[u8],
+        ) -> bool;
+
+        type Sha512Hasher;
+
+        fn new_sha512_hasher() -> Box<Sha512Hasher>;
+        fn sha512_update(self: &mut Sha512Hasher, chunk: &[u8]);
+        fn sha512_finish(self: &mut Sha512Hasher) -> Vec<u8>;
+    }
+}
+
+struct IntegrityBlockResult {
+    block: Option<CxxIntegrityBlock>,
+    error: Option<crate::ParseError>,
+}
+
+struct CxxIntegrityBlock {
+    size: usize,
+    web_bundle_id: String,
+    attributes_cbor: Vec<u8>,
+    signature_stack: Vec<CxxSignatureStackEntry>,
+}
+
+struct CxxSignatureStackEntry {
+    attributes_cbor: Vec<u8>,
+    signature_type: crate::SignatureType,
+    public_key: Vec<u8>,
+    signature: Vec<u8>,
+}
+
+fn parse_integrity_block_ffi(input: &[u8]) -> Box<IntegrityBlockResult> {
+    match parse_integrity_block(input) {
+        Ok(block) => Box::new(IntegrityBlockResult {
+            block: Some(CxxIntegrityBlock {
+                size: block.size,
+                web_bundle_id: block.web_bundle_id.to_owned(),
+                attributes_cbor: block.attributes_cbor,
+                signature_stack: block
+                    .signature_stack
+                    .into_iter()
+                    .map(|entry| CxxSignatureStackEntry {
+                        attributes_cbor: entry.attributes_cbor,
+                        signature_type: entry.signature_type,
+                        public_key: entry.public_key.to_vec(),
+                        signature: entry.signature.to_vec(),
+                    })
+                    .collect(),
+            }),
+            error: None,
+        }),
+
+        Err(error) => Box::new(IntegrityBlockResult {
+            block: None,
+            error: Some(error),
+        }),
+    }
+}
+
+fn err(result: &IntegrityBlockResult) -> bool {
+    result.error.is_some()
+}
+
+fn unwrap(result: &mut IntegrityBlockResult) -> Box<CxxIntegrityBlock> {
+    Box::new(result.block.take().expect("called unwrap on an error"))
+}
+
+fn error_message(result: &IntegrityBlockResult) -> String {
+    result
+        .error
+        .as_ref()
+        .expect("called error_message on a successful result")
+        .message
+        .clone()
+}
+
+fn error_is_version_error(result: &IntegrityBlockResult) -> bool {
+    result
+        .error
+        .as_ref()
+        .expect("called error_is_version_error on a successful result")
+        .is_version_error
+}
+
+fn size(block: &CxxIntegrityBlock) -> usize {
+    block.size
+}
+
+fn web_bundle_id(block: &CxxIntegrityBlock) -> String {
+    block.web_bundle_id.clone()
+}
+
+fn attributes_cbor(block: &CxxIntegrityBlock) -> Vec<u8> {
+    block.attributes_cbor.clone()
+}
+
+fn signature_stack_size(block: &CxxIntegrityBlock) -> usize {
+    block.signature_stack.len()
+}
+
+fn signature_stack_entry(
+    block: &CxxIntegrityBlock,
+    index: usize,
+) -> Box<CxxSignatureStackEntry> {
+    Box::new(
+        block
+            .signature_stack
+            .get(index)
+            .expect("signature stack index out of bounds")
+            .clone(),
+    )
+}
+
+fn attributes_cbor(entry: &CxxSignatureStackEntry) -> Vec<u8> {
+    entry.attributes_cbor.clone()
+}
+
+fn signature_type(entry: &CxxSignatureStackEntry) -> SignatureType {
+    match entry.signature_type {
+        crate::SignatureType::Ed25519 => SignatureType::Ed25519,
+        crate::SignatureType::EcdsaP256SHA256 => SignatureType::EcdsaP256SHA256,
+        crate::SignatureType::Unknown => SignatureType::Unknown,
+    }
+}
+
+fn public_key(entry: &CxxSignatureStackEntry) -> Vec<u8> {
+    entry.public_key.clone()
+}
+
+fn signature(entry: &CxxSignatureStackEntry) -> Vec<u8> {
+    entry.signature.clone()
+}
+
+fn new_sha512_hasher() -> Box<Sha512Hasher> {
+    Box::new(Sha512Hasher::new())
+}
+
+fn sha512_update(hasher: &mut Sha512Hasher, chunk: &[u8]) {
+    hasher.update(chunk);
+}
+
+fn sha512_finish(hasher: &mut Sha512Hasher) -> Vec<u8> {
+    hasher.finish().to_vec()
+}
+
+impl Clone for CxxSignatureStackEntry {
+    fn clone(&self) -> Self {
+        Self {
+            attributes_cbor: self.attributes_cbor.clone(),
+            signature_type: self.signature_type,
+            public_key: self.public_key.clone(),
+            signature: self.signature.clone(),
+        }
+    }
+}
diff -urN a/third_party/blink/renderer/platform/BUILD.gn b/third_party/blink/renderer/platform/BUILD.gn
--- a/third_party/blink/renderer/platform/BUILD.gn	2026-10-05 19:00:24.347515521 +0200
+++ b/third_party/blink/renderer/platform/BUILD.gn	2026-10-05 20:00:17.899793062 +0200
@@ -1794,7 +1794,6 @@
     ":allow_discouraged_type",
     ":blink_platform_public_deps",
     ":platform_export",
-    "//build/rust/crubit",
     "//gpu/command_buffer/client:raster_interface",
     "//media/capture:capture_lib",
     "//mojo/public/cpp/base",
@@ -1821,8 +1820,8 @@
     "//ui/native_theme/features",
   ]
   deps = [
+    "fonts:font_format",
     ":rustfft_ffi",
-    "fonts:font_format_bindings",
     "//base:base_static",
     "//base/allocator:buildflags",
     "//build:chromecast_buildflags",
diff -urN a/third_party/blink/renderer/platform/fonts/BUILD.gn b/third_party/blink/renderer/platform/fonts/BUILD.gn
--- a/third_party/blink/renderer/platform/fonts/BUILD.gn	2026-10-05 19:01:22.133719863 +0200
+++ b/third_party/blink/renderer/platform/fonts/BUILD.gn	2026-10-05 19:05:55.167620594 +0200
@@ -7,10 +7,7 @@
 rust_static_library("font_format") {
   crate_root = "opentype/format_check.rs"
   sources = [ crate_root ]
-  cpp_api_from_rust = {
-    target_name = "font_format_bindings"
-    cpp_namespace = "font_format"
-  }
+  cxx_bindings = [ crate_root ]
   deps = [
     "//third_party/rust/font_types/v0_12:lib",
     "//third_party/rust/read_fonts/v0_43:lib",
diff -urN a/third_party/blink/renderer/platform/fonts/opentype/font_format_check.cc b/third_party/blink/renderer/platform/fonts/opentype/font_format_check.cc
--- a/third_party/blink/renderer/platform/fonts/opentype/font_format_check.cc	2026-10-05 19:02:04.843654823 +0200
+++ b/third_party/blink/renderer/platform/fonts/opentype/font_format_check.cc	2026-10-05 19:07:26.500422077 +0200
@@ -7,6 +7,7 @@
 #include "base/containers/span.h"
 #include "base/containers/span_rust.h"
 #include "base/numerics/byte_conversions.h"
+#include "third_party/blink/renderer/platform/fonts/opentype/format_check.rs.h"
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/wtf/vector.h"
 #include "third_party/skia/include/core/SkTypeface.h"
@@ -14,37 +15,35 @@
 namespace blink {
 
 FontFormatCheck::FontFormatCheck(sk_sp<SkData> sk_data)
-    : format_info_(font_format::get_font_format_info(
+    : format_info_(font_format_check::get_font_format_info(
           base::SpanToRustSlice(sk_data->byteSpan()))) {}
 
-FontFormatCheck::~FontFormatCheck() = default;
-
 bool FontFormatCheck::IsVariableFont() const {
-  return font_format::is_variable(format_info_);
+  return font_format_check::is_variable(*format_info_);
 }
 
 bool FontFormatCheck::IsCbdtCblcColorFont() const {
-  return font_format::is_cbdt_cblc(format_info_);
+  return font_format_check::is_cbdt_cblc(*format_info_);
 }
 
 bool FontFormatCheck::IsEbdtEblcMonochromeFont() const {
-  return font_format::is_ebdt_eblc(format_info_);
+  return font_format_check::is_ebdt_eblc(*format_info_);
 }
 
 bool FontFormatCheck::IsColrCpalColorFontV0() const {
-  return font_format::is_colrv0(format_info_);
+  return font_format_check::is_colrv0(*format_info_);
 }
 
 bool FontFormatCheck::IsColrCpalColorFontV1() const {
-  return font_format::is_colrv1(format_info_);
+  return font_format_check::is_colrv1(*format_info_);
 }
 
 bool FontFormatCheck::IsSbixColorFont() const {
-  return font_format::is_sbix(format_info_);
+  return font_format_check::is_sbix(*format_info_);
 }
 
 bool FontFormatCheck::IsCff2OutlineFont() const {
-  return font_format::is_cff2(format_info_);
+  return font_format_check::is_cff2(*format_info_);
 }
 
 bool FontFormatCheck::IsVariableColrV0Font() const {
@@ -58,7 +57,7 @@
 
 bool FontFormatCheck::IsAvar2Font() const {
   return RuntimeEnabledFeatures::FontFormatAvar2Enabled() &&
-         font_format::is_avar2(format_info_);
+         font_format_check::is_avar2(*format_info_);
 }
 
 FontFormatCheck::VariableFontSubType FontFormatCheck::ProbeVariableFont(
diff -urN a/third_party/blink/renderer/platform/fonts/opentype/font_format_check.h b/third_party/blink/renderer/platform/fonts/opentype/font_format_check.h
--- a/third_party/blink/renderer/platform/fonts/opentype/font_format_check.h	2026-10-05 19:03:01.288307862 +0200
+++ b/third_party/blink/renderer/platform/fonts/opentype/font_format_check.h	2026-10-05 19:05:55.207528852 +0200
@@ -5,7 +5,7 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_PLATFORM_FONTS_OPENTYPE_FONT_FORMAT_CHECK_H_
 #define THIRD_PARTY_BLINK_RENDERER_PLATFORM_FONTS_OPENTYPE_FONT_FORMAT_CHECK_H_
 
-#include "third_party/blink/renderer/platform/fonts/font_format.h"
+#include "third_party/blink/renderer/platform/fonts/opentype/format_check.rs.h"
 #include "third_party/blink/renderer/platform/platform_export.h"
 #include "third_party/blink/renderer/platform/wtf/allocator/allocator.h"
 #include "third_party/skia/include/core/SkData.h"
@@ -19,7 +19,7 @@
 
  public:
   explicit FontFormatCheck(sk_sp<SkData>);
-  virtual ~FontFormatCheck();
+  virtual ~FontFormatCheck() = default;
   virtual bool IsVariableFont() const;
   virtual bool IsCbdtCblcColorFont() const;
   virtual bool IsEbdtEblcMonochromeFont() const;
@@ -46,7 +46,7 @@
   enum class COLRVersion { kCOLRV0, kCOLRV1, kNoCOLR };
 
  private:
-  font_format::FontFormatInfo format_info_;
+  rust::Box<font_format_check::FontFormatInfo> format_info_;
 };
 
 }  // namespace blink
diff -urN a/third_party/blink/renderer/platform/fonts/opentype/format_check.rs b/third_party/blink/renderer/platform/fonts/opentype/format_check.rs
--- a/third_party/blink/renderer/platform/fonts/opentype/format_check.rs	2026-10-05 19:04:46.216403238 +0200
+++ b/third_party/blink/renderer/platform/fonts/opentype/format_check.rs	2026-10-05 19:05:55.207833908 +0200
@@ -23,7 +23,7 @@
     format_flags: Option<FontFormatFlags>,
 }
 
-pub fn get_font_format_info(font_bytes: &[u8]) -> FontFormatInfo {
+pub fn get_font_format_info(font_bytes: &[u8]) -> Box<FontFormatInfo> {
     let file_ref = make_font_ref_internal(font_bytes, 0);
 
     match file_ref {
@@ -32,11 +32,11 @@
                 font.table_directory().table_records().iter().map(|e| e.tag()).collect();
             let color_version = get_colr_version(&font);
             let avar_version = get_avar_version(&font);
-            FontFormatInfo {
+            Box::new(FontFormatInfo {
                 format_flags: Some(FontFormatFlags { table_tags, color_version, avar_version }),
-            }
+            })
         }
-        _ => FontFormatInfo::default(),
+        _ => Box::new(FontFormatInfo::default()),
     }
 }
 
@@ -44,10 +44,10 @@
     Some(font_ref.colr().ok()?.version())
 }
 
-pub fn is_colrv1(format_info: &FontFormatInfo) -> bool {
+fn is_colrv1(format_info: &FontFormatInfo) -> bool {
     matches!(&format_info.format_flags, Some(FontFormatFlags { color_version: Some(1), .. }),)
 }
-pub fn is_colrv0(format_info: &FontFormatInfo) -> bool {
+fn is_colrv0(format_info: &FontFormatInfo) -> bool {
     matches!(&format_info.format_flags, Some(FontFormatFlags { color_version: Some(0), .. }),)
 }
 
@@ -56,7 +56,7 @@
     Some((version.major, version.minor))
 }
 
-pub fn is_avar2(format_info: &FontFormatInfo) -> bool {
+fn is_avar2(format_info: &FontFormatInfo) -> bool {
     matches!(&format_info.format_flags, Some(FontFormatFlags { avar_version: Some((2, _)), .. }),)
 }
 
@@ -69,22 +69,39 @@
     }
 }
 
-pub fn is_variable(format_info: &FontFormatInfo) -> bool {
+fn is_variable(format_info: &FontFormatInfo) -> bool {
     has_tags(format_info, &[Tag::new(b"fvar")])
 }
 
-pub fn is_sbix(format_info: &FontFormatInfo) -> bool {
+fn is_sbix(format_info: &FontFormatInfo) -> bool {
     has_tags(format_info, &[Tag::new(b"sbix")])
 }
 
-pub fn is_cbdt_cblc(format_info: &FontFormatInfo) -> bool {
+fn is_cbdt_cblc(format_info: &FontFormatInfo) -> bool {
     has_tags(format_info, &[Tag::new(b"CBDT"), Tag::new(b"CBLC")])
 }
 
-pub fn is_ebdt_eblc(format_info: &FontFormatInfo) -> bool {
+fn is_ebdt_eblc(format_info: &FontFormatInfo) -> bool {
     has_tags(format_info, &[Tag::new(b"EBDT"), Tag::new(b"EBLC")])
 }
 
-pub fn is_cff2(format_info: &FontFormatInfo) -> bool {
+fn is_cff2(format_info: &FontFormatInfo) -> bool {
     has_tags(format_info, &[Tag::new(b"CFF2")])
 }
+
+#[cxx::bridge(namespace = "font_format_check")]
+pub mod ffi {
+    extern "Rust" {
+        type FontFormatInfo;
+
+        fn get_font_format_info(font_bytes: &[u8]) -> Box<FontFormatInfo>;
+        fn is_colrv1(format_info: &FontFormatInfo) -> bool;
+        fn is_colrv0(format_info: &FontFormatInfo) -> bool;
+        fn is_cbdt_cblc(format_info: &FontFormatInfo) -> bool;
+        fn is_ebdt_eblc(format_info: &FontFormatInfo) -> bool;
+        fn is_variable(format_info: &FontFormatInfo) -> bool;
+        fn is_sbix(format_info: &FontFormatInfo) -> bool;
+        fn is_cff2(format_info: &FontFormatInfo) -> bool;
+        fn is_avar2(format_info: &FontFormatInfo) -> bool;
+    }
+}
