--- a/build/config/rust.gni	2026-10-03 00:14:51.608115414 +0200
+++ b/build/config/rust.gni	2026-10-03 00:19:06.642341029 +0200
@@ -63,6 +63,16 @@
   # the bindgen exectuable).
   rust_bindgen_root = "//third_party/rust-toolchain"
 
+  # Directory under which to find one of `libclang.{dll,so}` (a `lib[64]` or
+  # `bin` directory containing the libclang shared library).
+  # We don't need to worry about multlib, but specify the full path here
+  # in case a distribution does.
+  if (host_os == "win") {
+    bindgen_libclang_path = "//third_party/rust-toolchain/bin"
+  } else {
+    bindgen_libclang_path = "//third_party/rust-toolchain/lib"
+  }
+
   # If you're using a Rust toolchain as specified by rust_sysroot_absolute,
   # set this to the output of `rustc -V`. Changing this string will cause all
   # Rust targets to be rebuilt, which allows you to update your toolchain and
--- a/build/rust/rust_bindgen.gni	2026-10-03 00:16:41.096038355 +0200
+++ b/build/rust/rust_bindgen.gni	2026-10-03 00:21:59.309195657 +0200
@@ -17,16 +17,13 @@
   _bindgen_path = "${_bindgen_path}.exe"
 }
 
-# On Windows, the libclang.dll is beside the bindgen.exe, otherwise it is in
-# ../lib.
-_libclang_path = rust_bindgen_root
-if (!use_chromium_rust_toolchain &&
-    (current_cpu == "ppc64" || current_cpu == "s390x")) {
-  _libclang_path = rust_sysroot_absolute + "/lib64"
-} else if (host_os == "win") {
-  _libclang_path += "/bin"
+if (clang_base_path != default_clang_base_path && custom_toolchain == "//build/toolchain/linux/unbundle:default") {
+  # Assume that the user has set this up properly, including handling multilib
+  _clang_libpath = clang_base_path + "/lib/clang/" + clang_version + "/include"
+  _clang_ld_libpath  = bindgen_libclang_path
 } else {
-  _libclang_path += "/lib64"
+  _clang_libpath = clang_base_path + "/lib/clang/" + clang_version
+  _clang_ld_libpath = clang_base_path + "/lib"
 }
 
 # Template to build Rust/C bindings with bindgen.
--- a/build/rust/rust_bindgen_generator.gni	2026-10-03 00:16:41.096063614 +0200
+++ b/build/rust/rust_bindgen_generator.gni	2026-10-03 00:33:34.846197760 +0200
@@ -169,7 +169,7 @@
       "--output",
       rebase_path(output_file, root_build_dir),
       "--libclang-path",
-      rebase_path(_libclang_path, root_build_dir),
+      rebase_path(bindgen_libclang_path, root_build_dir),
     ]
 
     if (_wrap_static_fns) {
@@ -233,14 +233,28 @@
     # says the wrong thing. We point it to our clang's resource dir which will
     # make it behave consistently with our other command line flags and allows
     # system headers to be found.
-    clang_resource_dir =
-        rebase_path(clang_base_path + "/lib64/clang/" + clang_version,
+    if (clang_base_path != default_clang_base_path && custom_toolchain == "//build/toolchain/linux/unbundle:default") {
+      clang_resource_dir =
+            rebase_path(clang_base_path + "/lib/clang/" + clang_version + "/include", root_build_dir)
+    } else {
+       clang_resource_dir =
+            rebase_path(clang_base_path + "/lib/clang/" + clang_version,
                     root_build_dir)
+    }
     args += [
       "-resource-dir",
       clang_resource_dir,
     ]
 
+    if (custom_toolchain == "//build/toolchain/linux/unbundle:default") {
+      # We need to pass the path to the libstdc++ headers to bindgen so that it
+      # can find them when parsing C++ headers.
+      args += [
+        "-I",
+        rebase_path(clang_base_path + "/lib/clang/" + clang_version + "/include/", root_build_dir),
+      ]
+    }
+
     if (is_win) {
       # On Windows we fall back to using system headers from a sysroot from
       # depot_tools. This is negotiated by python scripts and the result is
