diff -uar chromium-37.0.2062.94.orig/net//url_request/url_request.cc chromium-37.0.2062.94/net//url_request/url_request.cc
--- chromium-37.0.2062.94.orig/net//url_request/url_request.cc	2014-08-20 21:02:28.000000000 +0200
+++ chromium-37.0.2062.94/net//url_request/url_request.cc	2014-08-30 16:57:11.719344396 +0200
@@ -705,9 +705,6 @@
   if (referrer_policy_ ==
           CLEAR_REFERRER_ON_TRANSITION_FROM_SECURE_TO_INSECURE &&
       GURL(referrer_).SchemeIsSecure() && !url().SchemeIsSecure()) {
-#if !defined(OFFICIAL_BUILD)
-    LOG(FATAL) << "Trying to send secure referrer for insecure load";
-#endif
     referrer_.clear();
     base::RecordAction(
         base::UserMetricsAction("Net.URLRequest_StartJob_InvalidReferrer"));
