--- ./clang/lib/Driver/ToolChains/Gnu.cpp.orig	2021-08-03 07:58:39.000000000 +0200
+++ ./clang/lib/Driver/ToolChains/Gnu.cpp	2021-08-24 01:08:18.022153891 +0200
@@ -2081,6 +2081,13 @@
   static const char *const ARMLibDirs[] = {"/lib"};
   static const char *const ARMTriples[] = {"arm-linux-gnueabi"};
   static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
+                                             "arm-slackware-linux-gnueabihf",
+                                             "armv6-slackware-linux-gnueabihf",
+                                             "armv6l-slackware-linux-gnueabihf",
+                                             "armv6hl-slackware-linux-gnueabihf",
+                                             "armv7-slackware-linux-gnueabihf",
+                                             "armv7l-slackware-linux-gnueabihf",
+                                             "armv7hl-slackware-linux-gnueabihf",
                                              "armv7hl-redhat-linux-gnueabi",
                                              "armv6hl-suse-linux-gnueabi",
                                              "armv7hl-suse-linux-gnueabi"};
@@ -2105,6 +2112,7 @@
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
+      "i486-slackware-linux", "i586-slackware-linux", "i686-slackware-linux",
       "i586-linux-gnu",      "i686-linux-gnu",        "i686-pc-linux-gnu",
       "i386-redhat-linux6E", "i686-redhat-linux",     "i386-redhat-linux",
       "i586-suse-linux",     "i686-montavista-linux", "i686-gnu",
