--- chromium.SlackBuild.orig	2026-08-07 16:18:07.974705079 +0200
+++ chromium.SlackBuild	2026-08-08 03:27:40.781073429 +0200
@@ -1,5 +1,5 @@
 #!/bin/bash
-# $Id: chromium.SlackBuild,v 1.576 2026/08/05 14:52:51 root Exp root $
+# $Id: chromium.SlackBuild,v 1.543 2026/03/13 18:13:29 root Exp root $
 # Copyright 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024, 2025, 2026  Eric Hameleers, Eindhoven, NL
 # All rights reserved.
 #
@@ -23,9 +23,10 @@
 #
 # -----------------------------------------------------------------------------
 #
-# Slackware SlackBuild script 
+# Slackware SlackBuild script
 # ===========================
 # By:              Eric Hameleers <alien@slackware.com>
+# Modified by      Martin Östman (Minime) <slackware[at]slackwarelinux[dot]se>
 # For:             chromium and chromium-ungoogled
 # Descr:           Open Source version of Chrome Web Browser
 # URL:             http://www.chromium.org/
@@ -40,9 +41,9 @@
 #                  - /usr/bin/python symlink pointing to python3.x
 #                  - java runtime
 #                  - Internet access (for downloading toolchain bins)
-# Needs:       
+# Needs:
 # Old Changelog:    chromium.buildhistory
-# Changelog:   
+# Changelog:
 # 142.0.7444.134-1: 06/nov/2025 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
 # 142.0.7444.162-1: 15/nov/2025 by Eric Hameleers <alien@slackware.com>
@@ -84,16 +85,28 @@
 #                   * Update to M146.
 # 146.0.7680.75-1:  13/mar/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
+#                   14/mar/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * As there are issues compiling rust for ARM32.
+#                     Removed all ARM32 build fixes.
 # 146.0.7680.80-1:  14/mar/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
+#                   16/mar/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * Fixed build for AArch64
+#                   * Added fix for rust bytemuck error
 # 146.0.7680.153-1: 18/mar/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
 # 146.0.7680.164-1: 24/mar/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
+#                   31/mar/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * Fixed build for AArch64
 # 146.0.7680.177-1: 01/apr/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
+#                   07/apr/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * Fixed build for AArch64
 # 147.0.7727.55-1:  08/apr/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update to M147.
+#                   14/apr/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * Fixed build for AArch64
 # 147.0.7727.101-1: 16/apr/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
 # 147.0.7727.116-1: 23/apr/2026 by Eric Hameleers <alien@slackware.com>
@@ -126,12 +139,15 @@
 #                   * Update to M151.
 # 151.0.7922.75-1:  04/aug/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
-# 
+# 151.0.7922.108-1:  07/aug/2026 by Minime <slackware[at]slackwarelinux[dot]se>
+#                   * Update.
+#                   * Fixed build for AArch64
+#
 # -----------------------------------------------------------------------------
-# 
+#
 # Run 'sh chromium.SlackBuild' to build a Slackware package.
 # The package (.txz) and .txt file as well as build logs are created in /tmp .
-# Install it using 'installpkg'. 
+# Install it using 'installpkg'.
 #
 # -----------------------------------------------------------------------------
 
@@ -148,10 +164,10 @@
 
 SRCNAM=chromium
 PRGNAM=${SRCNAM}${SRCEXT}     # chromium, chromium-dev, chromium-ungoogled, etc
-VERSION=${VERSION:-151.0.7922.75}
-UNGOOGLED=${UNGOOGLED:-151.0.7922.75-1}
+VERSION=${VERSION:-151.0.7922.108}
+UNGOOGLED=${UNGOOGLED:-151.0.7922.108-1}
 BUILD=${BUILD:-1}
-TAG=${TAG:-alien}
+TAG=${TAG:-SLse}
 
 # It's recommended to build 32bit binaries in a 32bit chroot on a 64bit OS.
 # Cross-compiling 32bit binaries on 64bit Slackware; set to "YES":
@@ -273,7 +289,7 @@
 fi
 
 # Slackware branding:
-BUILDER="by alienBOB for Slackware"
+BUILDER="by Minime for Slackware"
 
 # Where do we look for sources?
 SRCDIR=$(cd $(dirname $0); pwd)
@@ -293,7 +309,7 @@
 # If you rebuild this package, you can use your own API keys. See:
 #   http://www.chromium.org/developers/how-tos/api-keys
 # This browser is now compiled without API keys. This will disable stuff like
-# browser sync and other functionality which accesses Google APIs. 
+# browser sync and other functionality which accesses Google APIs.
 # The following variables must be set to access Google's public APIs:
 #   _google_api_key
 #   _google_default_client_id
@@ -518,7 +534,7 @@
 touch $OUTPUT/patch-${PRGNAM}.log
 # Arch contributions:
 echo "-- chromium_widevine_arm.patch --"
-cat $SRCDIR/patches/chromium_widevine_arm.patch \
+cat $SRCDIR/patches/arch/enable-widevine-arm64.patch \
   | patch -p1 --verbose \
   2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
 echo "-- chromium147_revert-clang-no-lifetime-dse-flag.patch --"
@@ -587,9 +603,9 @@
 #  2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
 # Dealing with the error:
 # TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
-# Fix building using Python 3.9 on Slackware 15.0 by placing 
+# Fix building using Python 3.9 on Slackware 15.0 by placing
 # the future import right above the first actual module import:
-sed -i ./tools/metrics/histograms/extract_histograms.py \
+ sed -i ./tools/metrics/histograms/extract_histograms.py \
   -e "/^import bisect/i from __future__ import annotations\n"
 # Chromium 151's copy of tools/metrics/BUILD.gn appears to be taken in the
 # midst of a multi-stage rafactor. Fix by adding the missing second part
@@ -928,7 +944,7 @@
     "ozone_platform_wayland=${do_wayland}"
     #"rtc_use_pipewire=${do_wayland}"
     #"rtc_link_pipewire=${do_wayland}"
-    #'rtc_pipewire_version="0.3"' # Otherwise chromium tries to link absent 0.2 
+    #'rtc_pipewire_version="0.3"' # Otherwise chromium tries to link absent 0.2
     'proprietary_codecs=true'
     'remove_webcore_debug_symbols=true'
     'symbol_level=0'
@@ -1276,7 +1292,7 @@
   fi
   cd third_party/ffmpeg
     ## To prevent error on x86_64:
-    ## "clang-5.0: error: invalid linker name in argument '-fuse-ld=lld'" 
+    ## "clang-5.0: error: invalid linker name in argument '-fuse-ld=lld'"
     ## in third_party/ffmpeg/build.x64.linux/ChromeOS/ffbuild/config.log
     #sed -e "s/--extra-ldflags=-fuse-ld=lld/--extra-ldflags=''/" \
     #    -i chromium/scripts/build_ffmpeg.py
@@ -1416,9 +1432,13 @@
 export AS=as
 
 # Dawn needs our system go:
-rm -rf third_party/dawn/tools/golang/linux-amd64
-#ln -s ${GOROOT} third_party/dawn/tools/golang/linux-${SYSROOT_ARCH}
-ln -s ${GOROOT} third_party/dawn/tools/golang/linux-amd64
+if [ "$ARCH" == "aarch64" ]; then
+  rm -rf third_party/dawn/tools/golang/linux-${SYSROOT_ARCH}
+  ln -s ${GOROOT} third_party/dawn/tools/golang/linux-${SYSROOT_ARCH}
+else
+  rm -rf third_party/dawn/tools/golang/linux-amd64
+  ln -s ${GOROOT} third_party/dawn/tools/golang/linux-amd64
+fi
 
 echo "-- Generating ninja build files."
 out/Release/gn gen out/Release -v --args="${_chromium_conf[*]}" \
@@ -1518,11 +1538,11 @@
 # Install appdata:
 # This file provides the 'process_template' function:
 if [ -f chrome/installer/linux/common/installer.include ]; then
-  source chrome/installer/linux/common/installer.include 
+  source chrome/installer/linux/common/installer.include
 else
-  source remoting/host/installer/linux/installer.include 
+  source remoting/host/installer/linux/installer.include
 fi
-source chrome/installer/linux/common/chromium-browser.info 
+source chrome/installer/linux/common/chromium-browser.info
 
 mkdir -p $PKG/usr/share/metainfo
 if [ $USE_UNGOOGLED -eq 0 ]; then
@@ -1591,6 +1611,17 @@
 install -D -m0644 out/Release/${PRGNAM}.desktop \
   $PKG/usr/share/applications/${PRGNAM}.desktop
 
+# Fix user-agent for aarch64
+if [ "$ARCH" == "aarch64" ]; then
+  MAJOR_VER="$(echo ${VERSION} | cut -d. -f1)"
+  cp out/Release/${PRGNAM}.desktop out/Release/${PRGNAM}-netflix.desktop
+  sed -i out/Release/${PRGNAM}-netflix.desktop \
+    -e "s/Name\=Chromium-ungoogled Web Browser/Name\=Chromium-ungoogled Netflix/" \
+    -e "s/\/${PRGNAM}/\/${PRGNAM} \-\-user-agent=\"Mozilla\/5.0 \(X11; CrOS ${ARCH} 10895.56.0\) AppleWebKit\/537.36 \(KHTML, like Gecko\) Chrome\/${MAJOR_VER}.0.0.0 Safari\/537.36\"/"
+  install -D -m0644 out/Release/${PRGNAM}-netflix.desktop \
+    $PKG/usr/share/applications/${PRGNAM}-netflix.desktop
+fi
+
 # Install desktop icons:
 for size in 16 32; do
   install -D -m0644 \
@@ -1738,5 +1769,12 @@
 cd -
 cat $PKG/install/slack-desc | grep "^${PRGNAM}" > $OUTPUT/${PRGNAM}-${VERSION}-${PKGARCH}-${BUILD}${TAG}.txt
 
+# Cleanup
+if [ "${GOOGLE_CLANG}" == "src" ] && [  ${BUILD_CLANG} -eq 1 ]; then
+  if  [ -f ${MYCLANG_PATH}/ld ]; then
+    rm ${MYCLANG_PATH}/ld
+  fi
+fi
+
 # Restore the original umask:
 umask ${_UMASK_}
