--- chromium.SlackBuild.114.106.orig	2023-06-06 19:05:06.000000000 +0200
+++ chromium.SlackBuild	2023-06-11 15:35:33.628061040 +0200
@@ -1,5 +1,4 @@
 #!/bin/bash
-# $Id: chromium.SlackBuild,v 1.340 2023/06/06 17:04:56 root Exp root $
 # Copyright 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023  Eric Hameleers, Eindhoven, NL
 # All rights reserved.
 #
@@ -21,421 +20,115 @@
 #   OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 #   SUCH DAMAGE.
 #
+# Modified by Martin Östman (Minime) for slackware ARM (SARPI).
+#
 # -----------------------------------------------------------------------------
 #
-# Slackware SlackBuild script 
+# Slackware SlackBuild script
 # ===========================
 # By:              Eric Hameleers <alien@slackware.com>
-# For:             chromium
+# Modified by      Martin Östman (Minime) <slackware[at]slackwarelinux[dot]se>
+# For:             chromium/chromium-ungoogled
 # Descr:           Open Source version of Chrome Web Browser
 # URL:             http://www.chromium.org/
 # Build needs:     - ninja;
-#                  - nodejs >= 18.7.0
+#                  - nodejs >= 18.7.0;
 #                  - gcc >= 8 , llvm >= 16 for bootstrapping gn;
 #                  - nasm >= 2.14 (yasm no longer being used);
-#                  - cmake >= 3.24.1
+#                  - cmake >= 3.13.4;
 #                  - python3 >= 3.7, python3-setuptools;
 #                  - java runtime
 #                  - Internet access (for downloading toolchain bins)
-# Needs:       
-# Changelog:   
-# 31.0.1650.57-1:  18/nov/2013 by Eric Hameleers <alien@slackware.com>
-#                  * Initial build.
-# 31.0.1650.63-1:  18/Nov/2013 by Eric Hameleers <alien@slackware.com>
-# 31.0.1650.67-1:  19/dec/2013 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 32.0.1700.77-1:  14/jan/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 32.0.1700.102-1: 29/jan/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 32.0.1700.107-1: 04/feb/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 33.0.1750.117-1: 21/feb/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 33.0.1750.146-1: 04/mar/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 33.0.1750.149-1: 11/mar/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 33.0.1750.152-1: 18/mar/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 34.0.1847.116-1: 09/apr/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 34.0.1847.132-1: 25/apr/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 34.0.1847.137-1: 14/may/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 35.0.1916.114-1: 20/may/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 35.0.1916.153-1: 10/jun/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-# 36.0.1985.125-1: 16/jun/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with several security fixes.
-#                    We can no longer download toolchain binaries in advance,
-#                    so the build will now require internet connectivity.
-# 36.0.1985.143-1: 12/aug/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes, Chrome also comes with an
-#                    updated Flash plugin: 14.0.0.177.
-# 37.0.2062.94-1:  27/aug/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes.
-#                    New: an embedded PDF viewer.
-# 37.0.2062.120-1: 09/sep/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes, Chrome also comes with an
-#                    updated Flash plugin: 15.0.0.152.
-# 38.0.2125.101-1: 08/oct/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes.
-# 38.0.2125.104-1: 18/oct/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes. Chrome also updates its
-#                    Flash plugin to 15.0.0.189.
-# 38.0.2125.111-1: 28/oct/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 38.0.2125.122-1: 11/nov/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes. Chrome also updates its
-#                    Flash plugin to 15.0.0.223.
-# 39.0.2171.65-1:  17/nov/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes.  See
-#                    http://googlechromereleases.blogspot.nl/2014/11/stable-channel-update_18.html
-# 39.0.2171.71-1:  26/nov/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes. Chrome also updates its
-#                    Flash plugin to 15.0.0.239.
-# 39.0.2171.95-1:  09/dec/2014 by Eric Hameleers <alien@slackware.com>
-#                  * Update comes with security fixes. Chrome also updates its
-#                    Flash plugin to 16.0.0.235.
-# 39.0.2171.95-2:  01/jan/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Add support for the WIdevine CDM plugin (i.e. Netflix playback)
-# 39.0.2171.99-1:  14/jan/2015 by Eric Hameleers <alien@slackware.com>
-#                  * New release accompanied by pepperflash 16.0.0.257. 
-# 40.0.2214.91-1:  21/jan/2015 by Eric Hameleers <alien@slackware.com>
-#                  * New release.
-# 40.0.2214.111-1: 06/feb/2015 by Eric Hameleers <alien@slackware.com>
-#                  * New release.
-# 41.0.2272.76-1:  05/mar/2015 by Eric Hameleers <alien@slackware.com>
-#                  * First release of Chromium 41 stable.
-# 41.0.2272.118-1: 02/apr/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 42.0.2311.90-1:  15/apr/2015 by Eric Hameleers <alien@slackware.com>
-#                  * First release of Chromium 42 stable.
-# 42.0.2311.135-1: 30/apr/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 43.0.2357.65-1:  20/may/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Security update, first 43 version.
-# 43.0.2357.81-1:  29/may/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 43.0.2357.124-1: 09/jun/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 43.0.2357.125-1: 12/jun/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 43.0.2357.125-2: 19/jun/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Disable "OK Google" hotwording completely.
-# 43.0.2357.130-1: 22/jun/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 43.0.2357.130-2: 25/jun/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Fix broken password manager in KDE.
-# 43.0.2357.132-1: 08/jul/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 43.0.2357.134-1: 15/jul/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 44.0.2403.89-1:  22/jul/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update - first 44 release.
-# 44.0.2403.107-1: 25/jul/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update - emergency bugfix release.
-# 44.0.2403.155-1: 11/aug/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 44.0.2403.157-1: 21/aug/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 45.0.2454.85-1:  02/sep/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update - first 45 release.
-# 45.0.2454.93-1:  16/sep/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 45.0.2454.101-1: 25/sep/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 46.0.2490.71-1:  14/oct/2015 by Eric Hameleers <alien@slackware.com>
-#                  * First 46 release.
-# 46.0.2490.86-1:  22/nov/2015 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 47.0.2526.73-1:  03/dec/2015 by Eric Hameleers <alien@slackware.com>
-#                  * First 47 release.
-# 48.0.2564.82-1:  24/jan/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 48 release.
-# 48.0.2564.97-1:  31/jan/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 48.0.2564.109-1: 13/feb/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 49.0.2623.75-1:  05/mar/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 49 release.  Binary plugins such as widevine and
-#                    pepperflash are no longer available for 32-bit since
-#                    Google stopped providing binaries for 32-bit Chrome.
-# 49.0.2623.108-1: 25/feb/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 49.0.2623.112-1: 08/apr/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 50.0.2661.75-1:  14/apr/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 50 release.
-# 50.0.2661.94-1:  29/apr/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 50.0.2661.102-1: 12/may/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 51.0.2704.63-1:  26/may/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 51 release.
-# 51.0.2704.84-1:  07/jun/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 51.0.2704.103-1: 17/jun/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 52.0.2743.82-1:  24/jul/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 52.0.2743.116-1: 04/aug/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 53.0.2785.92-1:  06/sep/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 53 release.
-# 53.0.2785.92-2:  12/sep/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Apply patch required for Slackware 14.1.
-# 54.0.2840.90-1:  03/nov/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 54 release. Build system changed from gyp to gn.
-# 54.0.2840.100-1: 17/nov/2016 by Eric Hameleers <alien@slackware.com>
-#                  * Updated. Security fix. Removed the VAAPI patch because
-#                    several people have issues with it.
-# 55.0.2883.75-1:  12/dec/2016 by Eric Hameleers <alien@slackware.com>
-#                  * First 55 release.
-# 56.0.2924.76-1:  30/jan/2017 by Eric Hameleers <alien@slackware.com>
-#                  * First 56 release.
-# 56.0.2924.76-2:  25/feb/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Built for Slackware 14.1 with some additional patches.
-# 57.0.2987.98-1:  14/mar/2017 by Eric Hameleers <alien@slackware.com>
-#                  * First 57 release.
-# 58.0.3029.96-1:  04/may/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 58.
-# 59.0.3071.86-1:  06/jun/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 59.
-# 60.0.3112.78-1:  25/jun/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 60.
-# 61.0.3163.79-1:  11/sep/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 61.
-# 61.0.3163.100-1: 25/sep/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 62.0.3202.62-1:  18/oct/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 62.
-# 62.0.3202.75-1:  27/oct/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update, switch compiler to clang instead of gcc.
-# 62.0.3202.94-1:  25/nov/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 63.0.3239.108-1: 17/dec/2017 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 63.
-# 63.0.3239.132-1: 07/jan/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 64.0.3282.119-1: 26/jan/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 64.
-# 64.0.3282.119-2: 29/jan/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Enable HEVC support.
-# 64.0.3282.140-1: 05/feb/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 64.0.3282.186-1: 06/mar/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 65.0.3325.146-1: 09/mar/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M65.
-# 65.0.3325.162-1: 16/mar/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 65.0.3325.181-1: 31/mar/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 66.0.3359.117-1: 25/apr/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M66.
-# 66.0.3359.139-1: 25/apr/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Security update (CVE-2018-6118).
-# 66.0.3359.181-1: 18/may/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 67.0.3396.62-1:  31/may/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M67.
-# 67.0.3396.79-1:  08/jun/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 67.0.3396.99-1:  02/jul/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 68.0.3440.84-1:  01/aug/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 68.
-# 69.0.3497.81-1:  05/sep/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 69.
-# 69.0.3497.100-1: 28/sep/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update, trying to fix pulseaudio.
-# 70.0.3538.67-1:  24/oct/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 70.
-# 70.0.3538.102-1: 11/nov/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 71.0.3578.80-1:  05/dec/2018 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M71.
-# 72.0.3626.81-1:  31/jan/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M72.
-# 72.0.3626.121-1: 03/mar/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to 72.0.3626.121.
-# 73.0.3683.75-1:  15/mar/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M73.
-# 73.0.3683.75-2:  17/mar/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Fix 32bit build.
-# 73.0.3683.86-1:  31/mar/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 73.0.3683.103-1: 14/apr/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 74.0.3729.108-1: 24/apr/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M74.
-# 74.0.3729.169-1: 22/may/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 75.0.3770.80-1:  08/jun/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M75.
-# 75.0.3770.90-1:  13/jun/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 75.0.3770.100-1: 20/jun/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 75.0.3770.100-2: 20/jun/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Fix HEVC support.
-# 75.0.3770.142-1: 15/jul/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 76.0.3809.87-1:  31/jul/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M76.
-# 76.0.3809.132-1: 30/aug/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 77.0.3865.75-1:  11/sep/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M77.
-# 77.0.3865.90-1:  11/sep/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 77.0.3865.120-1: 10/oct/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 78.0.3904.70-1:  10/oct/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M78.
-# 78.0.3904.87-1:  01/nov/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Critical security update.
-# 78.0.3904.97-1:  11/nov/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 78.0.3904.108-1: 19/nov/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 79.0.3945.79-1:  11/dec/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 79.0.3945.88-1:  18/dec/2019 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 79.0.3945.117-1: 08/jan/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 79.0.3945.130-1: 19/jan/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Security update.
-# 80.0.3987.87-1:  05/feb/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M80.
-# 80.0.3987.106-1: 14/feb/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 80.0.3987.122-1: 27/feb/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 80.0.3987.132-1: 04/mar/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 80.0.3987.149-1: 18/mar/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 81.0.4044.92-1:  08/apr/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M81.
-# 81.0.4044.113-1: 16/apr/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 81.0.4044.122-1: 25/apr/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 81.0.4044.129-1: 29/apr/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 81.0.4044.138-1: 06/may/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 83.0.4103.61-1:  20/may/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M83.
-# 83.0.4103.97-1:  07/jun/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 83.0.4103.106-1: 16/jun/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 83.0.4103.116-1: 23/jun/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-#                    Add support for VAAPI. This is off by default but
-#                    can be enabled by adding '--ignore-gpu-blacklist' and
-#                    then one of '--use-gl=desktop' or '--use-gl=egl'.
-#                    See https://chromium.googlesource.com/chromium/src/+/master/docs/gpu/vaapi.md#VaAPI-on-Linux
-#                    for the intended usage (Intel VAAPI and MESA VAAPI).
-#                    To support Nvidia and other GPUs, patches would be needed.
-# 84.0.4147.89-1:  20/jul/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M84.
-# 84.0.4147.94-1:  23/jul/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated. Use a newer nodejs to make the sources compile
-#                    on 32bit Slackware 14, plus this also fixes the
-#                    broken development tools.
-# 84.0.4147.105-1: 01/aug/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 85.0.4183.83-1:  28/aug/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M85.
-# 85.0.4183.102-1: 09/sep/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 85.0.4183.121-1: 22/sep/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 86.0.4240.75-1:  07/oct/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M86.
-# 86.0.4240.111-1: 22/oct/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 86.0.4240.183-1: 03/nov/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 86.0.4240.198-1: 13/nov/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 87.0.4280.66-1:  23/nov/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M87.
-# 87.0.4280.88-1:  03/dec/2020 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 87.0.4280.141-1: 07/jan/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 88.0.4324.96-1:  21/jan/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M88.
-# 88.0.4324.146-1: 04/feb/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 88.0.4324.150-1: 06/feb/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 88.0.4324.182-1: 17/feb/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 88.0.4324.190-1: 23/feb/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 89.0.4389.72-1:  03/mar/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M89.
-# 89.0.4389.82-1:  07/mar/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 89.0.4389.90-1:  13/mar/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 89.0.4389.114-1: 31/mar/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 89.0.4389.128-1: 14/apr/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 90.0.4430.72-1:  15/apr/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Update to M90.
-# 90.0.4430.93-1:  28/apr/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 90.0.4430.212-1: 11/may/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Updated.
-# 91.0.4472.101-1: 11/jun/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
-# 91.0.4472.114-1: 19/jun/2021 by Eric Hameleers <alien@slackware.com>
-#                  * Update.
+# Needs:
+# Changelog:
 # 92.0.4515.107-1: 22/jul/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M92.
 # 92.0.4515.131-1: 05/aug/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 92.0.4515.159-1: 17/aug/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  30/aug/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Added lates patchset "92.0.4515.x"
+#                  * Repatched clang.toolchains.arm_i586.triple for ARM.
+#                  * Fixed chromium_conf for ARM
+#                  * Fixed target triple for arm.
+#                  * Fixed ARM sysroot path to libsqlite3.so, glibconfig.h,
+#                    dbus-arch-deps.h, graphene-config.h and incorrect
+#                    conf file "(arch)-linux-gnu.conf" in ARM system.
+#                  * Fixed compiling google's internal clang for ARM.
+#                  31/aug/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Fixed missing link to debias readelf for ARM.
+#                  2/sep/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Alienbob thought the code is messy, and i agree.
+#                    Started on some code cleanup.
+#                  * Removed eu-strip patch and rebuild of eu-strip as this
+#                    not needed.
+#                  * Cleaned up duplicate entries in chromium config array.
 # 93.0.4577.63-1:  01/sep/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M93. Add some mods that hopefully make building
 #                    on ARM platforms easier.
 # 93.0.4577.82-1:  13/sep/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+# 93.0.4577.82-1:  20/sep/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update to M93.
+#                  * Added latest patchsets for chromium-93.0.4577.8
+#                  * Fixed incorrect/missing softlink
+#                  * Added softlink cleanup
+#                  * Fixed Widevine support patch for ARM
+#                    https://www.linuxquestions.org/questions/slackware-arm-108/how-to-get-widevine-4-10-2252-0-working-on-slackware-arm-4175700846/
+#                  22/sep/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Fixed custom clang option, set default
+#                    to systems clang version
+#                  27/sep/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Merged script with AlienBOB´s slackbuild
+#                  * Fixed and adjusted script to my llvm build.
+#                  * Note: There will be an issue with Slackware ARMs
+#                          llvm/clang version as its missing lld.
+#                          This is one reason i make a custom llvm/clang package.
+#                  * cleared changelog history as its kind of irrelevant and old...
 # 94.0.4606.71-1:  02/oct/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M94.
+# 94.0.4606.81-1:  12/oct/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update to M94.
+#                  * Needs llvm/clang 13, so system clang is no longer an option.
+#                  * Will build with python 3.9, but if upgraded to python 3.10
+#                    there will be issues. Need to revert package python 3.10 to 3.9.
 # 95.0.4638.69-1:  28/oct/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M95.
 # 96.0.4664.45-1:  15/nov/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M96.
+#                  06/dec/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Fixed chromium/ungoogled for ARM
 # 96.0.4664.93-1:  09/dec/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 96.0.4664.110-1: 14/dec/2021 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+# 96.0.4664.110-1  21/dec/2021 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * bump version to 96.0.4664.110.
 # 97.0.4692.71-1:  05/jan/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M97.
+# 97.0.4692.71-1:  09/jan/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * bump version to 97.0.4692.71.
 # 97.0.4692.99-1:  20/jan/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  27/jan/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
 # 98.0.4758.80-1:  02/feb/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M98.
-# 98.0.4758.102-1:  16/feb/2022 by Eric Hameleers <alien@slackware.com>
+#                  04/feb/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
+# 98.0.4758.102-1: 16/feb/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  17/feb/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
 # 99.0.4844.51-1:  02/mar/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M99.
+# 99.0.4844.51-1:  09/mar/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update to M99 fix chromium/ungoogled for ARM
 # 99.0.4844.74-1:  16/mar/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  17/mar/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
 # 99.0.4844.82-1:  24/mar/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 99.0.4844.84-1:  27/mar/2022 by Eric Hameleers <alien@slackware.com>
@@ -454,18 +147,40 @@
 #                  * Update.
 # 101.0.4951.64-1: 11/may/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  18/may/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
 # 102.0.5005.61-1: 25/may/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M102.
+#                  27/may/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM.
+# 102.0.5005.61-3: 02/jun/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Ungoogled update.
+#                  - Fix bypassing ssl errors not being possible anymore (see #1976)
+#                  - Properly bump revision to 3 (see #1980)
+#                  - Fix the browser crashing when adding a search engine entry (see #1970)
+#                  - Add two flags to allow looking more like the tor browser (see #1920)
 # 102.0.5005.115-1:10/jun/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  12/jun/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
+# 103.0.5060.53-1 :22/jun/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update to M103 version fix chromium/ungoogled for ARM
 # 103.0.5060.53-1: 25/jun/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M103.
+# 103.0.5060.114-1 :12/jul/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
+# 103.0.5060.134-1 :20/jul/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Bump version fix chromium/ungoogled for ARM
 # 103.0.5060.134-1:20/jul/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 104.0.5112.79-1: 02/aug/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M104.
+#                  05/aug/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update to M104 version fix chromium/ungoogled for ARM.
 # 104.0.5112.101-1:17/aug/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  19/aug/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update version fix chromium/ungoogled for ARM.
 # 105.0.5195.52-1: 02/aug/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M105.
 # 105.0.5195.102-1:04/aug/2022 by Eric Hameleers <alien@slackware.com>
@@ -486,6 +201,8 @@
 #                  * Update.
 # 107.0.5304.110-1: 09/nov/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                   15/nov/2022 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update version fix chromium/ungoogled for ARM. (Failed to build)
 # 107.0.5304.121-1: 25/nov/2022 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 108.0.5359.71-1: 02/dec/2022 by Eric Hameleers <alien@slackware.com>
@@ -504,47 +221,78 @@
 #                  * System libdrm on Slackware 14.2 is too old.
 # 109.0.5414.119-1: 25/jan/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  30/jan/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update version fix chromium/ungoogled for ARM.
 # 110.0.5481.77-1: 09/feb/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M110.
+#                  09/feb/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update M110, version fix chromium/ungoogled for ARM.
 # 110.0.5481.100-1:15/feb/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  19/feb/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update, version fix chromium/ungoogled for ARM.
 # 110.0.5481.177-1:26/feb/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  26/feb/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update, version fix chromium/ungoogled for ARM.
 # 111.0.5563.64-1: 09/mar/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M111.
+#                  09/mar/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 111.0.5563.110-1:23/mar/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  23/mar/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 111.0.5563.147-1:29/mar/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  01/apr/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 112.0.5615.49-1: 05/apr/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M112.
+#                  07/apr/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 112.0.5615.121-1:15/apr/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  16/apr/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 112.0.5615.165-1:20/apr/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  20/apr/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 113.0.5672.63-1: 03/may/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M113.
-# 113.0.5672.92-1: 15/may/2023 by Eric Hameleers <alien@slackware.com>
+#                  03/may/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
+# 113.0.5672.92-1: 10/may/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * Update.
+#                  * version fix chromium/ungoogled for ARM.
+#                  15/may/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 113.0.5672.126-1:16/may/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
+#                  18/may/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
 # 114.0.5735.45-1: 27/may/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update to M114.
 # 114.0.5735.90-1: 31/may/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
 # 114.0.5735.106-1:06/jun/2023 by Eric Hameleers <alien@slackware.com>
 #                  * Update.
-# 
+#                  07/jun/2023 by Minime <slackware[at]slackwarelinux[dot]se>
+#                  * version fix chromium/ungoogled for ARM.
+#
 # -----------------------------------------------------------------------------
-# 
+#
+# -----------------------------------------------------------------------------
+#
 # Run 'sh chromium.SlackBuild' to build a Slackware package.
 # The package (.txz) and .txt file as well as build logs are created in /tmp .
-# Install it using 'installpkg'. 
+# Install it using 'installpkg'.
 #
 # -----------------------------------------------------------------------------
 
-# Build an ungoogled chromium package (default no):
-USE_UNGOOGLED=${USE_UNGOOGLED:-0}
+# Build an ungoogled chromium package (default yes):
+USE_UNGOOGLED=${USE_UNGOOGLED:-1}
 
 # Build a development version of chromium (default no):
 USE_DEV=${USE_DEV:-0}
@@ -559,7 +307,7 @@
 VERSION=${VERSION:-114.0.5735.106}
 UNGOOGLED=${UNGOOGLED:-114.0.5735.106-1}
 BUILD=${BUILD:-1}
-TAG=${TAG:-alien}
+TAG=${TAG:-SLse}
 
 # Cross-compiling 32bit binaries on 64bit Slackware; set to "YES":
 CROSS32=${CROSS32:-"NO"}
@@ -632,15 +380,16 @@
 # clang for download0:
 USE_CLANG=${USE_CLANG:-1}
 
-# Build a custom clang or use the system clang?
-# CLANG_MIN is the minimal clang version needed to compile chromium,
-# see 'CLANG_REVISION' in ./tools/clang/scripts/update.py :
-CLANG_MIN=16
-if [ $(clang -dumpversion |cut -d. -f1) -ge ${CLANG_MIN} ]; then
-  BUILD_CLANG=${BUILD_CLANG:-0}
-else
-  BUILD_CLANG=${BUILD_CLANG:-1}
-fi
+# Use custom clang the systems clang version or if you installed the build i made
+# to save time rather than trying to compile googles clang build every time.
+USE_CUSTCLANG=${USE_CUSTCLANG:-1}
+
+# If set custom clang (USE_CUSTCLANG "1")
+# Make sure you to set correct clang base path.
+# Path where you installed clang, default is set to "/usr" (The system).
+# If you installed my llvm/clang package then set CLANG_BASE_PATH="/opt/llvm".
+#CLANG_BASE_PATH="/usr"
+CLANG_BASE_PATH="/opt/llvm"
 
 # If you don't want to enable support for Widevine CDM
 # (content decryption module) say "0" here.
@@ -664,6 +413,22 @@
 PKG=$TMP/package-$PRGNAM
 OUTPUT=${OUTPUT:-/tmp}
 
+if [ ${USE_CUSTCLANG} -eq 1 ] && [ ! -f "${CLANG_BASE_PATH}/bin/clang" ]; then
+  echo "You have not installed llvm/clang"
+  echo "Build will fail"
+  exit 1
+else
+  CLANG_MIN=16
+  if [ ! $(${CLANG_BASE_PATH}/bin/clang --version |grep '^clang' |cut -d' ' -f3 |cut -d. -f1) -ge ${CLANG_MIN} ]; then
+    echo "Need newer version of llvm/clang"
+    echo "Will proceed in 10 seconds and build internal llvm/clang"
+    echo "If you rather wish to install prebuilt llvm/clang, then abort by pressing Ctrl-C"
+    USE_CUSTCLANG=${USE_CUSTCLANG:-0}
+    sleep 10
+  fi
+  echo "Internal llvm/clang build started..."
+fi
+
 # Visible directory listing of anything Chrome related is:
 #   http://gsdview.appspot.com/
 # Official location (non-browsable) of the above is:
@@ -674,7 +439,7 @@
 # If you rebuild this package, you can use your own API keys. See:
 #   http://www.chromium.org/developers/how-tos/api-keys
 # This browser is now compiled without API keys. This will disable stuff like
-# browser sync and other functionality which accesses Google APIs. 
+# browser sync and other functionality which accesses Google APIs.
 # The following variables must be set to access Google's public APIs:
 #   _google_api_key
 #   _google_default_client_id
@@ -754,14 +519,16 @@
              #DSSE="0"
              DSSE="1"
              ;;
-  armv7*)    SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=hard"
+  armv7*)    SLKCFLAGS="-O2 -march=armv7-a -mtune=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard"
              SLKLDFLAGS="-Wl,-z,notext"; LIBDIRSUFFIX=""
              DSSE="1"
              # Widevine not supported (no binary library available):
-             USE_CDM=0
+             # However there are workarounds so why deny this?
+             # Added patch that enables widevine support for armv7
+             #USE_CDM=0
              # VAAPI not supported:
              USE_VAAPI=0
-             FFMPEG_ARCH="arm"
+             FFMPEG_ARCH="arm-neon"
              SYSROOT_ARCH="arm"
              SYSROOT_TARGET="arm-linux-gnueabihf"
              PKGARCH="$ARCH"
@@ -781,9 +548,8 @@
 esac
 
 case "$ARCH" in
-    armv7*)  TARGET=$ARCH-slackware-linux-gnueabi ;;
     arm*)    TARGET=${SYSROOT_ARCH}-slackware-linux-gnueabihf ;;
-    *)       TARGET=$ARCH-slackware-linux ;;
+    *)       TARGET=${ARCH}-slackware-linux ;;
 esac
 
 # Exit the script on errors:
@@ -902,14 +668,19 @@
 cat $SRCDIR/patches/chromium_protobuf_32bit.patch \
   | patch -p0 --verbose \
   2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
-
 # Be compatible with gcc 10:
 cat $SRCDIR/patches/chromium_gcc10.patch | patch -p1 --verbose \
   2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
+
+# Fixes widevine support for arm
+cat $SRCDIR/patches/widevine_support_for_arm.patch \
+  | patch -p1 --verbose 2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
+
 # Fix the link on 32bit:
 sed -e '/ldflags /s/-m32/&", "-Wl,-z,notext/' \
   -i build/config/compiler/BUILD.gn \
    2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
+#if [ "$ARCH" == "i586" ] || [[ "$ARCH" =~ arm.* ]]; then
 if [ "$ARCH" == "i586" ] ; then
   # Don't complain about unsupported flags when we add
   # "--disable-seccomp-sandbox" on 32bit OS:
@@ -945,7 +716,6 @@
   cat $SRCDIR/patches/chromium_ffmpeg_prop.patch | patch -p1 --verbose \
     2>&1 | tee -a $OUTPUT/patch-${PRGNAM}.log
 fi
-
 if [ ${USE_UNGOOGLED} -ne 0 ]; then
   # Apply ungoogled-chromium patch kit:
   UGDIR=$TMP/tmp-$PRGNAM/ungoogled-chromium-${UNGOOGLED}
@@ -968,18 +738,20 @@
 # And fix the 32bit compilation while we are at it:
 sed -i tools/clang/scripts/build.py -e "s/ldflags = \[\]/ldflags = \['-latomic'\]/"
 
-## Avoid failure because the built clang version is not what update.py expects:
-#sed -i tools/clang/scripts/update.py -e "s/RELEASE_VERSION = '16'/RELEASE_VERSION = '16.0.0'/"
-
-# ARM platform specifics:
+# If ARM, then try to force the build script to build clang for arm instead of x86
+# Note: This will most likely fail as it does not detect the correct target architecture.
+# One of many reasons i built chromium's llvm package separately.
 if [[ "$ARCH" =~ arm.* ]]; then
-  # Fix the hard-coded bootstrap target 'X86':
+  # We are compiling bootstrap for ARM and not X86.
   sed -i tools/clang/scripts/build.py \
-    -e "s/bootstrap_targets = 'X86'/bootstrap_targets = 'ARM'/"
-
-  # Compile only the relevant back-ends (ARM, AArch64 and X86):
+    -e "s/bootstrap_targets = \x27X86\x27/bootstrap_targets = \x27ARM\x27/"
+  # We are using ARM, so to speed things up we only build the relevant back-ends
+  # (ARM,AArch64 and x86), since it’s very unlikely that we will cross-compile to other arches.
+  # x86 is also needed for Compiler-RT tests.
+  # Old outdated chromium >= 102?
+  #-e "s/targets = \x27AArch64;ARM;Mips;PowerPC;SystemZ;WebAssembly;X86\x27/targets = \x27ARM;X86;AArch64\x27/"
   sed -i tools/clang/scripts/build.py \
-    -e "s/\(^ *targets = \).*$/&1'ARM;X86;AArch64'/"
+    -e "s/targets = \x27AArch64;ARM;Mips;PowerPC;RISCV;SystemZ;WebAssembly;X86\x27/targets = \x27ARM;X86;AArch64\x27/"
 fi
 
 # When compiling google's clang, use the system cmake
@@ -1025,7 +797,7 @@
 echo Building ...
 
 [ ${USE_CLANG} -eq 1 ] && do_clang=true || do_clang=false
-[ ${BUILD_CLANG} -eq 1 ] && do_custclang=true || do_custclang=false
+[ ${USE_CUSTCLANG} -eq 1 ] && do_clang_plug=false || do_clang_plug=true
 [ ${USE_CDM} -eq 1 ] && do_cdm=true || do_cdm=false
 [ ${USE_HOTWORD} -eq 1 ] && do_hotword=true || do_hotword=false
 [ ${USE_PA} -eq 1 ] && do_pa=true || do_pa=false
@@ -1033,7 +805,7 @@
 [ ${USE_SYSROOT} -eq 1 ] && do_sysroot=true || do_sysroot=false
 [ ${USE_VAAPI} -eq 1 ] && do_vaapi=true || do_vaapi=false
 [ ${USE_WAYLAND} -eq 1 ] && do_wayland=true || do_wayland=false
-[ ${FFMPEG_HEVC} -eq 1 ] && do_hevc=true || do_hevc=false 
+[ ${FFMPEG_HEVC} -eq 1 ] && do_hevc=true || do_hevc=false
 
 export LDFLAGS="$SLKLDFLAGS"
 export CXXFLAGS="$SLKCFLAGS -Wno-unknown-warning-option"
@@ -1053,46 +825,40 @@
     'blink_symbol_level=0'
     "enable_hevc_parser_and_hw_decoder=${do_hevc}"
     "enable_platform_hevc=${do_hevc}"
-    "enable_platform_hevc_decoding=${do_hevc}"
-    "enable_platform_encrypted_hevc=${do_hevc}"
     'enable_platform_ac3_eac3_audio=true'
     'enable_platform_mpeg_h_audio=true'
     'enable_platform_dolby_vision=true'
     'enable_mse_mpeg2ts_stream_parser=true'
     'enable_precompiled_headers=false'
     'enable_pseudolocales=false'
-    'enable_webrtc=true'
     'fatal_linker_warnings=false'
     "ffmpeg_branding=\"${FFMPEG_BRANDING}\""
     "is_clang=${do_clang}"
-    #"llvm_force_head_revision=${do_clang}"
+    "clang_use_chrome_plugins=${do_clang_plug}"
     'is_debug=false'
+    'is_official_build=true'
+    'dcheck_always_on=false'
     "is_component_build=${SHARED_FFMPEG}"
     "is_component_ffmpeg=${SHARED_FFMPEG}"
     "link_pulseaudio=${do_pa}"
-    "linux_sandbox_path=\"/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/chrome_sandbox\""
-    #'linux_strip_binary=true'
-    'linux_use_bundled_binutils=false'
-    "ozone_platform_wayland=${do_wayland}"
     #"rtc_use_pipewire=${do_wayland}"
     #"rtc_link_pipewire=${do_wayland}"
-    #'rtc_pipewire_version="0.3"' # Otherwise chromium tries to link absent 0.2 
+    #'rtc_pipewire_version="0.3"' # Otherwise chromium tries to link absent 0.2
+    "ozone_platform_wayland=${do_wayland}"
     'proprietary_codecs=true'
-    'remove_webcore_debug_symbols=true'
     'symbol_level=0'
     'treat_warnings_as_errors=false'
-    "usb_ids_path=\"/usr/share/hwdata/usb.ids\""
     'use_allocator="none"'
     'use_bundled_fontconfig=false'
     'use_cfi_icall=false' # https://crbug.com/866290
     'use_cups=true'
     'use_custom_libcxx=true'
-    'use_gconf=false'
     'use_kerberos=false'
     'use_lld=true'
     'use_ozone=true'
     "use_pulseaudio=${do_pa}"
     'use_system_libffi=false'
+    "use_system_libwayland=${do_wayland}"
     'use_system_minigbm=false'
     "use_qt=${do_qt}"
     "use_sysroot=${do_sysroot}"
@@ -1101,14 +867,7 @@
     "extra_cxxflags=\"${CXXFLAGS}\""
     "extra_ldflags=\"${LDFLAGS}\""
 )
-# Needed for Ungoogled Chromium 94 to prevent compile error related to
-# missing pref 'kGoogleServicesConsentedToSync' because 'is_debug=false' seems
-# to no longer suffice. Also, need to reduce crash proneness according to
-# https://groups.google.com/a/chromium.org/g/chromium-packagers/c/2E5AayFuFt4 :
-_chromium_conf+=(
-  'is_official_build=true'
-  'dcheck_always_on=false'
-)
+
 # Prevent segfaults with llvm-12.0.1 (Chromium 94), thanks Void Linux:
 _chromium_conf+=(
   'is_cfi=false'
@@ -1116,32 +875,54 @@
   'use_cfi_icall=false'
   'chrome_pgo_phase=0'
 )
-if [ ${USE_CLANG} -eq 1 ] && [  ${BUILD_CLANG} -eq 0 ]; then
-  # Use system toolchain:
+
+# Set base path custom clang
+if [ ${USE_CUSTCLANG} -eq 1 ]; then
   _chromium_conf+=(
-    'custom_toolchain="//build/toolchain/linux/unbundle:default"'
-    'host_toolchain="//build/toolchain/linux/unbundle:default"'
+    'clang_use_default_sample_profile=false'
+    "clang_base_path=\"${CLANG_BASE_PATH}\""
   )
 fi
+
+if [[ ! "$ARCH" =~ arm.* ]]; then
+  _chromium_conf+=(
+    "enable_platform_hevc_decoding=${do_hevc}"
+    "enable_platform_encrypted_hevc=${do_hevc}"
+    'enable_hidpi=true'
+    'enable_webrtc=true'
+    "linux_sandbox_path=\"/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/chrome_sandbox\""
+    #'linux_strip_binary=true'
+    'linux_use_bundled_binutils=false'
+    'remove_webcore_debug_symbols=true'
+    "usb_ids_path=\"/usr/share/hwdata/usb.ids\""
+    'use_gconf=false'
+  )
+fi
+
 if [ ${USE_UNGOOGLED} -ne 0 ]; then
   # Add ungoogled flags if requested:
   readarray -t -O ${#_chromium_conf[@]} _chromium_conf < ${UGDIR}/flags.gn
 else
-  # Add remainder of standard chromium flags:
-  # https://chromium.googlesource.com/chromium/src/+/refs/heads/main/testing/variations/README.md
+  if [[ ! "$ARCH" =~ arm.* ]]; then
+    # Add remainder of standard chromium flags:
+    _chromium_conf+=(
+        "enable_hotwording=${do_hotword}"
+        'enable_pepper_cdms=true'
+    )
+  fi
   _chromium_conf+=(
-    "clang_use_chrome_plugins=${do_custclang}"
-    'enable_hangout_services_extension=true'
-    "enable_hotwording=${do_hotword}"
-    'enable_nacl=false'
-    'enable_nacl_nonsfi=false'
-    'enable_pepper_cdms=true'
-    "enable_widevine=${do_cdm}"
-    'exclude_unwind_tables=true'
-    'disable_fieldtrial_testing_config=true'
-    'logging_like_official_build=true'
+      'disable_fieldtrial_testing_config=true'
+      'logging_like_official_build=true'
+      'enable_hangout_services_extension=true'
+      'enable_js_type_check=true'
+      'enable_nacl=false'
+      'enable_nacl_nonsfi=false'
+      "enable_widevine=${do_cdm}"
+      'exclude_unwind_tables=true'
+      'fieldtrial_testing_like_official_build=true'
   )
 fi
+
 # Add Google API key if defined:
 if [ -n "$_google_api_key" ]; then
   _chromium_conf+=(
@@ -1161,11 +942,11 @@
   _chromium_conf+=( 'use_system_libdrm=false' )
 fi
 
-#if [ "$ARCH" == "i586" ] || [ "${CROSS32}" = "YES" ]; then
-#  _chromium_conf+=( 'target_cpu="x86"' )
-#fi
 if [[ "$ARCH" =~ arm.* ]]; then
-  _chromium_conf+=( 'target_cpu="arm"' 'target_os="linux"' )
+  _chromium_conf+=(
+      'target_cpu="arm"'
+      'target_os="linux"'
+  )
 fi
 
     # Change this back when CRBUG 592268 is fixed (error during final link:
@@ -1173,46 +954,47 @@
     # See https://bugs.chromium.org/p/chromium/issues/detail?id=592268
     # icu_use_data_file_flag=false
 
-## The array of possible replacements of embedded with system libraries is
-## found in 'build/linux/unbundle/replace_gn_files.py':
-#gn_system_libraries=(
-#  # 'bzip2'
-#  # 'ffmpeg'
-#  # 'flac'
-#  # 'freetype2'
-#  # 'harfbuzz'
-#  # 'icu'
-#  # 'libdrm'
-#  # 'libevent'
-#  # 'libjpeg'
-#  # 'libpng'
-#  # 'libvpx'
-#  # 'libwebp'
-#  # 'libxml'
-#  # 'libxslt'
-#  # 'opus'
-#  # 're2'
-#  # 'snappy'
-#  # 'zlib'
-#  )
-#
-## Unbundle the libraries which we use from the system.
-## (taken from Arch Linux:)
-## This *should* do what the remove_bundled_libraries.py script does,
-## with the added benefit of not having to list all the remaining libraries:
-#echo "-- Unbundling system libraries from $PRGNAM sources."
-#for ULIB in ${gn_system_libraries[@]}; do
-#  find -type f -path "*third_party/$ULIB/*" \
-#    \! -path "*third_party/$ULIB/chromium/*" \
-#    \! -path "*third_party/$ULIB/google/*" \
-#    \! -regex '.*\.\(gn\|gni\|isolate\)' \
-#    -delete
-#done
-#
-## Change bundled libraries to system ones:
-#echo "-- Change bundled to system libraries in $PRGNAM sources."
-#python3 build/linux/unbundle/replace_gn_files.py --system-libraries ${gn_system_libraries[@]}
-#
+if [ ${FFMPEG_HEVC} -eq 0 ]; then
+  # The array of possible replacements of embedded with system libraries is
+  # found in 'build/linux/unbundle/replace_gn_files.py':
+  gn_system_libraries=(
+    # 'bzip2'
+    # 'ffmpeg'
+    'flac'
+    # 'freetype2'
+    # 'harfbuzz'
+    # 'icu'
+    # 'libdrm'
+    # 'libevent'
+    # 'libjpeg'
+    # 'libpng'
+    # 'libvpx'
+    # 'libwebp'
+    # 'libxml'
+    # 'libxslt'
+    # 'opus'
+    # 're2'
+    # 'snappy'
+    # 'zlib'
+    )
+
+  # Unbundle the libraries which we use from the system.
+  # (taken from Arch Linux:)
+  # This *should* do what the remove_bundled_libraries.py script does,
+  # with the added benefit of not having to list all the remaining libraries:
+  echo "-- Unbundling system libraries from $PRGNAM sources."
+  for ULIB in ${gn_system_libraries[@]}; do
+    find -type f -path "*third_party/$ULIB/*" \
+      \! -path "*third_party/$ULIB/chromium/*" \
+      \! -path "*third_party/$ULIB/google/*" \
+      \! -regex '.*\.\(gn\|gni\|isolate\)' \
+      -delete
+  done
+
+  # Change bundled libraries to system ones:
+  echo "-- Change bundled to system libraries in $PRGNAM sources."
+  python3 build/linux/unbundle/replace_gn_files.py --system-libraries ${gn_system_libraries[@]}
+fi
 
 # Do not make the build fail, see Chromium issue #386097:
 python3 third_party/libaddressinput/chromium/tools/update-strings.py \
@@ -1254,8 +1036,6 @@
     build/linux/${SYSROOTDIR}/usr/include/graphene-1.0/graphene-config.h
 
   # Take care of Debian/Ubuntu related linker issues:
-  #echo "/usr/lib/${SYSROOT_TARGET}/nss" >> \
-  #  build/linux/${SYSROOTDIR}/etc/ld.so.conf.d/${SYSROOT_TARGET}.conf
   for SOFILE in build/linux/${SYSROOTDIR}/usr/lib/${SYSROOT_TARGET}/nss/*.so ;
   do
     ln -sf nss/$(basename $SOFILE) \
@@ -1265,10 +1045,8 @@
     build/linux/${SYSROOTDIR}/usr/lib/${SYSROOT_TARGET}/libsqlite3.so
 
   # The platform-specific alias for the ARM readelf binary is missing
-  # from the debian sysroot image:
-  if [ ! -f build/linux/${SYSROOTDIR}/usr/bin/${SYSROOT_TARGET}-readelf ]; then
-    ln -s /usr/bin/readelf \
-      build/linux/${SYSROOTDIR}/usr/bin/${SYSROOT_TARGET}-readelf
+  if [ ! -f /usr/bin/${SYSROOT_TARGET}-readelf ]; then
+    ln -s /usr/bin/readelf /usr/bin/${SYSROOT_TARGET}-readelf
   fi
 fi
 
@@ -1283,26 +1061,12 @@
 # --- Compile --- #
 
 # Final link uses lots of file descriptors.
-ulimit -n 4096
+ulimit -n 2048
 
 # Compile google's internal clang sources if needed
-if [ ${USE_CLANG} -eq 1 ] && [  ${BUILD_CLANG} -eq 1 ]; then
+if [ ${USE_CLANG} -eq 1 ] && [ ${USE_CUSTCLANG} -eq 0 ]; then
   # Build clang from source instead of downloading:
   echo "-- Building google's clang toolchain."
-  HDR2REM=""
-  if [ "$ARCH" == "i586" ]; then
-    # Actually 32bit compilation fails by not finding 'bits/c++config.h'.
-    # I could only work around this using a kludge:
-    # cp -ia /usr/include/c++/$(gcc -dumpversion)/${TARGET}/bits/* /usr/include/c++/$(gcc -dumpversion)/bits/
-    # Keep a list of the symlinks we'll be creating:
-    for HDR in /usr/include/c++/$(gcc -dumpversion)/${TARGET}/bits/*.h ; do
-      if [ ! -f /usr/include/c++/$(gcc -dumpversion)/bits/$(basename $HDR) ];
-    then
-      ln -s $HDR /usr/include/c++/$(gcc -dumpversion)/bits/
-      HDR2REM="$(basename $HDR) $HDR2REM"
-      fi
-    done
-  fi
   LDFLAGS="$LDFLAGS" \
   python3 tools/clang/scripts/build.py \
     --bootstrap \
@@ -1326,7 +1090,11 @@
 
   # Ensure that only this new clang is used, ignoring the system version:
   MYCLANG_PATH="$TMP/tmp-$PRGNAM/${SRCNAM}-${VERSION}/third_party/llvm-build/Release+Asserts/bin"
-  export PATH="$MYCLANG_PATH:$PATH"
+  export PATH="${MYCLANG_PATH}:$PATH"
+elif [ ${USE_CLANG} -eq 1 ] && [ ${USE_CUSTCLANG} -eq 1 ]; then
+  # Ensure correct path for custom clang version:
+  MYCLANG_PATH="${CLANG_BASE_PATH}/bin"
+  export PATH="${MYCLANG_PATH}:$PATH"
 fi
 
 # For Chromium 92, disable ffmpeg re-configuration on 32bit OS,
@@ -1347,10 +1115,8 @@
   fi
   cd third_party/ffmpeg
     ## To prevent error on x86_64:
-    ## "clang-5.0: error: invalid linker name in argument '-fuse-ld=lld'" 
+    ## "clang-5.0: error: invalid linker name in argument '-fuse-ld=lld'"
     ## in third_party/ffmpeg/build.x64.linux/ChromeOS/ffbuild/config.log
-    #sed -e "s/--extra-ldflags=-fuse-ld=lld/--extra-ldflags=''/" \
-    #    -i chromium/scripts/build_ffmpeg.py
     # Actual reconfigure steps:
     python3 chromium/scripts/build_ffmpeg.py linux ${FFMPEG_ARCH} \
       --branding ${FFMPEG_BRANDING} -- ${FFMPEG_CUSTOM_ARGS} \
@@ -1374,27 +1140,17 @@
 # Generate the ninja build files:
 echo "-- Bootstrapping 'gn'."
 
-if [ ${USE_CLANG} -eq 1 ] && [  ${BUILD_CLANG} -eq 1 ]; then
-  # Symlink ld to lld to make gn linking stage succeed:
-  cd $MYCLANG_PATH ; ln -s lld ld ; cd - 1>/dev/null
+# Symlink ld to lld to make gn linking stage succeed:
+if [ "$USE_CLANG" -eq 1 ] && [ ! "$MYCLANG_PATH" = "/usr/bin" ]; then
+  if [ ! -f ${MYCLANG_PATH}/ld ]; then
+    cd ${MYCLANG_PATH} ; ln -s lld ld ; cd - 1>/dev/null
+  fi
 fi
 
 LDFLAGS="$LDFLAGS" python3 tools/gn/bootstrap/bootstrap.py \
   2>&1 | tee $OUTPUT/gyp-${PRGNAM}.log
   #--use-custom-libcxx \
 
-#if [ ${USE_CLANG} -eq 1 ] && [  ${BUILD_CLANG} -eq 1 ]; then
-#  # Remove symlink again:
-#  rm $MYCLANG_PATH/ld
-#fi
-
-if [ ${USE_CLANG} -eq 1 ] && [  ${BUILD_CLANG} -eq 1 ]; then
-  if [ -n "$HDR2REM" ]; then
-    # Remove the temporary header symlinks we created earlier:
-    ( cd /usr/include/c++/$(gcc -dumpversion)/bits/ && rm $HDR2REM)
-  fi
-fi
-
 echo "-- Generating ninja build files."
 out/Release/gn gen out/Release -v --args="${_chromium_conf[*]}" \
   2>&1 | tee -a $OUTPUT/gyp-${PRGNAM}.log
@@ -1402,29 +1158,28 @@
 # Compile everything:
 echo "-- Compiling the lot."
 
-if [ ${USE_CLANG} -eq 1 ] && [ ${BUILD_CLANG} -eq 1 ]; then
+if [ "$USE_CLANG" -eq 1 ] && [ ! "$MYCLANG_PATH" = "/usr/bin" ]; then
   # Use the clang that we build in-tree instead of using the system version:
-  echo "-- Using our custom clang."
   export CC="$MYCLANG_PATH/clang"
   export CXX="$MYCLANG_PATH/clang++"
   export LD="$MYCLANG_PATH/lld"
   export AR="$MYCLANG_PATH/llvm-ar"
   export NM="$MYCLANG_PATH/llvm-nm"
-elif [ "$USE_CLANG" = "0" ]; then
+  export AS="$MYCLANG_PATH/llvm-as"
+elif [ "$USE_CLANG" -eq 1 ] && [ "$MYCLANG_PATH" = "/usr/bin" ]; then
+  # Use the clang that we build in-tree instead of using the system version:
+  export CC="clang"
+  export CXX="clang++"
+  export LD="lld"
+  export AR="ar"
+  export NM="nm"
+  export AS="as"
+else
   # Be careful not to override CC and CXX (may have been set by gcc5):
-  echo "-- Using gcc/g++ compilers."
   export CC=${CC:-"gcc"}
   export CXX=${CXX:-"g++"}
-else
-  # Use the system clang:
-  echo "-- Using the system clang."
-  export CC="/usr/bin/clang"
-  export CXX="/usr/bin/clang++"
-  export LD="/usr/bin/lld"
-  export AR="/usr/bin/llvm-ar"
-  export NM="/usr/bin/llvm-nm"
+  export AS=${AS:-"as"}
 fi
-export AS=as
 
 if [ $USE_UNGOOGLED -eq 1 ]; then
   # Incorrect build order triggers: "fatal error: 'components/feed/core/proto/v2/wire/chrome_feed_response_metadata.pb.h' file not found",
@@ -1576,6 +1331,13 @@
 # Add a desktop file:
 process_template chrome/installer/linux/common/desktop.template \
   out/Release/${PRGNAM}.desktop
+
+# Fix user-agent for ARM
+if [[ "$ARCH" =~ arm.* ]]; then
+  sed -i out/Release/${PRGNAM}.desktop \
+    -e "s/\/${PRGNAM}/\/${PRGNAM} \-\-user-agent=\"Mozilla\/5.0 \(X11; CrOS armv7l 10895.56.0\) AppleWebKit\/537.36 \(KHTML, like Gecko\) Chrome\/${VERSION} Safari\/537.36\"/"
+fi
+
 install -D -m0644 out/Release/${PRGNAM}.desktop \
   $PKG/usr/share/applications/${PRGNAM}.desktop
 
@@ -1725,5 +1487,15 @@
 cd -
 cat $PKG/install/slack-desc | grep "^${PRGNAM}" > $OUTPUT/${PRGNAM}-${VERSION}-${PKGARCH}-${BUILD}${TAG}.txt
 
+# Cleanup
+if [ "$USE_CLANG" -eq 1 ] && [ ! "$MYCLANG_PATH" = "/usr/bin" ]; then
+  if  [ -f ${MYCLANG_PATH}/ld ]; then
+    rm ${MYCLANG_PATH}/ld
+  fi
+fi
+if [ -f /usr/bin/${SYSROOT_TARGET}-readelf ]; then
+  rm /usr/bin/${SYSROOT_TARGET}-readelf
+fi
+
 # Restore the original umask:
 umask ${_UMASK_}
