--- chromium.SlackBuild.orig	2026-09-18 22:03:56.000000000 +0200
+++ chromium.SlackBuild.154.orig	2026-09-27 10:17:38.000000000 +0200
@@ -1,5 +1,5 @@
 #!/bin/bash
-# $Id: chromium.SlackBuild,v 1.590 2026/09/15 20:34:04 root Exp root $
+# $Id: chromium.SlackBuild,v 1.592 2026/09/25 18:59:29 root Exp root $
 # Copyright 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024, 2025, 2026  Eric Hameleers, Eindhoven, NL
 # All rights reserved.
 #
@@ -144,6 +144,8 @@
 #                   * Update to M153.
 # 153.0.8010.52-1:  18/sep/2026 by Eric Hameleers <alien@slackware.com>
 #                   * Update.
+# 154.0.8037.57-1:  25/sep/2026 by Eric Hameleers <alien@slackware.com>
+#                   * Update to M154.
 # 
 # -----------------------------------------------------------------------------
 # 
@@ -166,8 +168,8 @@
 
 SRCNAM=chromium
 PRGNAM=${SRCNAM}${SRCEXT}     # chromium, chromium-dev, chromium-ungoogled, etc
-VERSION=${VERSION:-153.0.8010.52}
-UNGOOGLED=${UNGOOGLED:-153.0.8010.52-1}
+VERSION=${VERSION:-154.0.8037.57}
+UNGOOGLED=${UNGOOGLED:-154.0.8037.57-1}
 BUILD=${BUILD:-1}
 TAG=${TAG:-alien}
 
@@ -1574,6 +1576,29 @@
 fi
 source chrome/installer/linux/common/chromium-browser.info 
 
+# This step used to be done for us, now we need to do it manually. We use
+# the 'process_template' function from an earlier sourced script:
+APPSTREAM_SCREENSHOT_URL=""
+appstream_screenshot_url=""
+DESKTOP_EXEC=${PRGNAM}
+desktop_exec=${PRGNAM}
+DESKTOP_ICON=${PRGNAM}
+desktop_icon=${PRGNAM}
+EXTRA_DESKTOP_ENTRIES=""
+extra_desktop_entries=""
+MENUNAME="${PRGNAM^} Web Browser"
+menuname="${PRGNAM^} Web Browser"
+PACKAGE=${PRGNAM}
+package=${PRGNAM}
+PROGNAME=${PRGNAM}
+progname=${PRGNAM}
+StartupWMClass="${PRGNAM}"
+startup_wm_class="${PRGNAM}"
+URI_SCHEME="x-scheme-handler/${PRGNAM}"
+uri_scheme="x-scheme-handler/${PRGNAM}"
+USR_BIN_SYMLINK_NAME=${PRGNAM}
+usr_bin_symlink_name=${PRGNAM}
+
 mkdir -p $PKG/usr/share/metainfo
 if [ $USE_UNGOOGLED -eq 0 ]; then
   process_template chrome/installer/linux/common/appdata.xml.template \
@@ -1614,23 +1639,6 @@
 # Add resource files:
 cp -a out/Release/resources $PKG/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/
 
-# This step used to be done for us, now we need to do it manually. We use
-# the 'process_template' function from an earlier sourced script:
-EXTRA_DESKTOP_ENTRIES=""
-extra_desktop_entries=""
-MENUNAME="${PRGNAM^} Web Browser"
-menuname="${PRGNAM^} Web Browser"
-PACKAGE=${PRGNAM}
-package=${PRGNAM}
-PROGNAME=${PRGNAM}
-progname=${PRGNAM}
-StartupWMClass="${PRGNAM}"
-startup_wm_class="${PRGNAM}"
-URI_SCHEME=""
-uri_scheme=""
-USR_BIN_SYMLINK_NAME=${PRGNAM}
-usr_bin_symlink_name=${PRGNAM}
-
 # Install a man page:
 process_template chrome/app/resources/manpage.1.in out/Release/${PRGNAM}.1
 install -D -m0644 out/Release/${PRGNAM}.1 $PKG/usr/man/man1/${PRGNAM}.1
